File: src/libs/ThirdParty.VBSuppliers/Properties/SunHotels/ISunHotelsSettings.vb
Imports ThirdParty.Abstractions

Public Interface ISunHotelsSettings
    ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property EmailAddress(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Language(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Currency(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property SupplierReference(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property SearchURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property BookURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property CancelURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, isMandotory As Boolean) As Integer
    ReadOnly Property Nationality(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property PreBookURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property AccommodationTypes(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property RequestPackageRates(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property HotelRequestLimit(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
End Interface

File: src/libs/iVectorOne.Suppliers/SunHotels/ISunHotelsSettings.cs
namespace iVectorOne.CSSuppliers.SunHotels
{
    public interface ISunHotelsSettings
    {
        string Password(IThirdPartyAttributeSearch tpAttributeSearch);
        string User(IThirdPartyAttributeSearch tpAttributeSearch);
        string ContactEmail(IThirdPartyAttributeSearch tpAttributeSearch);
        string LanguageCode(IThirdPartyAttributeSearch tpAttributeSearch);
        string Currency(IThirdPartyAttributeSearch tpAttributeSearch);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch);
        string SupplierReference(IThirdPartyAttributeSearch tpAttributeSearch);
        bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch);
        string SearchURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string BookingURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string CancellationURL(IThirdPartyAttributeSearch tpAttributeSearch);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool isMandotory);
        string CustomerCountryCode(IThirdPartyAttributeSearch tpAttributeSearch);
        string PrebookURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string AccommodationTypes(IThirdPartyAttributeSearch tpAttributeSearch);
        bool RequestPackageRates(IThirdPartyAttributeSearch tpAttributeSearch);
        int HotelBatchLimit(IThirdPartyAttributeSearch tpAttributeSearch);
    }
}